File: coreui/src/main/java/com/mbg/module/ui/fragment/BaseFragment.java
package com.mbg.module.ui.fragment;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.IdRes;
import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;

import com.jaeger.library.StatusBarUtil;
import com.mbg.module.ui.activity.BaseFragmentActivity;


public abstract class BaseFragment extends Fragment {
    public View mRootView;
    private int mContainer;


    public void showContent(Class<? extends BaseFragment> fragmentClass) {
        showContent(fragmentClass, null);
    }

    public void showContent(Class<? extends BaseFragment> fragmentClass, Bundle bundle) {
        BaseFragmentActivity activity = (BaseFragmentActivity) getActivity();
        if (activity != null) {
            activity.showContent(fragmentClass, bundle);
        }
    }

    public void setStatusBarColor(int color,int statusBarAlpha){
        StatusBarUtil.setColor(getActivity(),color,statusBarAlpha);
    }

    @LayoutRes
    protected abstract int onRequestLayout();

    protected abstract void initView();

    public final  <T extends View> T findViewById(@IdRes int id) {
        return mRootView.findViewById(id);
    }

    @Nullable
    @Override
    public final View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                                   @Nullable Bundle savedInstanceState) {
        mRootView=super.onCreateView(inflater,container,savedInstanceState);
        int id = onRequestLayout();
        if (id > 0&&mRootView==null) {
            mRootView = inflater.inflate(id, container, false);
            initView();
        }
        if (mRootView == null) {
            mRootView = onCreateView(savedInstanceState);
            initView();
        }else if (mRootView.getParent()!=null){
            ((ViewGroup) mRootView.getParent()).removeView(mRootView);
        }
        if (interceptTouchEvents()) {
            if (mRootView != null) {
                mRootView.setOnTouchListener(new View.OnTouchListener() {
                    @Override
                    public boolean onTouch(View view, MotionEvent motionEvent) {
                        return true;
                    }
                });
            }
        }
        return mRootView;
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        tryGetContainerId();
        try {
            if (view.getContext() instanceof Activity) {
                ((Activity) view.getContext()).getWindow().getDecorView().requestLayout();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void tryGetContainerId() {
        if (mRootView != null) {
            View parent = (View) mRootView.getParent();
            if (parent != null) {
                mContainer = parent.getId();
            }
        }
    }

    protected View onCreateView(Bundle savedInstanceState) {
        return mRootView;
    }

    protected boolean interceptTouchEvents() {
        return false;
    }

    public int getContainer() {
        if (mContainer == 0) {
            tryGetContainerId();
        }
        return mContainer;
    }

    public boolean onBackPressed() {
        return false;
    }

    public void finish() {
        BaseFragmentActivity activity = (BaseFragmentActivity) getActivity();
        if (activity != null) {
            activity.doBack(this);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }
}

File: coreui/src/main/java/com/mbg/module/ui/kotlin/fragment/BaseFragment.kt
package com.mbg.module.ui.kotlin.fragment

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment
import com.jaeger.library.StatusBarUtil
import com.mbg.module.ui.kotlin.activity.BaseFragmentActivity

abstract class BaseFragment : Fragment() {
    var mRootView: View? = null
    private var mContainer = 0

    @JvmOverloads
    fun showContent(fragmentClass: Class<out BaseFragment?>?, bundle: Bundle? = null) {
        val activity = activity as BaseFragmentActivity?
        activity?.showContent(fragmentClass!!, bundle)
    }

    fun setStatusBarColor(color: Int, statusBarAlpha: Int) {
        StatusBarUtil.setColor(activity, color, statusBarAlpha)
    }


    //返回Layout Id
    @LayoutRes
    abstract fun onRequestLayout(): Int

    //返回View
    open fun onCreateView(savedInstanceState: Bundle?): View? {
        return mRootView
    }

    abstract fun initView()
    fun <T : View?> findViewById(@IdRes id: Int): T {
        return mRootView!!.findViewById(id)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        tryGetContainerId()
        try {
            if (view.context is Activity) {
                (view.context as Activity).window.decorView.requestLayout()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mRootView = super.onCreateView(inflater, container, savedInstanceState)
        val id = onRequestLayout()
        if (id > 0 && mRootView == null) {
            mRootView = inflater.inflate(id, container, false)
            initView()
        }
        if (mRootView == null) {
            mRootView = onCreateView(savedInstanceState)
            initView()
        } else if (mRootView!!.parent != null) {
            (mRootView!!.parent as ViewGroup).removeView(mRootView)
        }
        if (interceptTouchEvents()) {
            mRootView?.setOnTouchListener{ view, motionEvent -> true
            }
        }
        return mRootView
    }

    private fun tryGetContainerId() {
        if (mRootView != null) {
            val parent = mRootView!!.parent as View
            mContainer = parent.id
        }
    }

    open fun interceptTouchEvents(): Boolean {
        return false
    }

    val container: Int
        get() {
            if (mContainer == 0) {
                tryGetContainerId()
            }
            return mContainer
        }

    fun onBackPressed(): Boolean {
        return false
    }

    fun finish() {
        val activity = activity as BaseFragmentActivity?
        activity?.doBack(this)
    }
}